components/rhs_thread/rhs_thread.jsx

```diff
@@ -73,11 +74,17 @@ export default class RhsThread extends React.Component {
 
         const openTime = (new Date()).getTime();
 
+        let loading = false;
+        if (this.props.posts.length < (Utils.getRootPost(this.props.posts).reply_count + 1)) {
+            loading = true;
+        }
+
```

Comment:
so this will not do 'lazy' loading, right? it will wait until all posts are available?
how will that work in this scenario:
1. message is posted
2. reply is posted
3. some time passes, lots of other stuff in the channel
4. another reply is posted
5. reload the channel, open rhs on the last reply
6. the store will only have the root post and the last reply (unless I am mistaken), so the loading spinner will stay until you scroll up in the main channel and fetch the first reply

Reply:
What it does is check if you have already everything in the store, so... if your root post have a reply count of X and the posts shown in the thread are less than X, it will request the whole thread, bot only at that time.

Reply:
I see.
What will happen for a thread where X = 100 and you only have root and the last post?
It will have to issue 100 requests?

Reply:
No you don't you get the whole thread directly.

Reply:
@jespino - can you elaborate? From your previous comment: "if your root post have a reply count of X and the posts shown in the thread are less than X, it will request the whole thread, bot **only at that time**"

Reply:
Yes, the idea is, don't load the whole thread of each post in the channel, you only load the posts that are in the channel plus the root posts. When you open a thread, I verify that you have already all the posts loaded (replyCount+1, must be the number of posts including root) if I have less than that, I load the thread, if not, I have already the whole thread loaded.